docs/proposals/policy.md

```diff
@@ -0,0 +1,258 @@
+
+`Hubert` needs wants to create a bot that only has permission to touch labels on DeploymentConfigs.  He can create a role like this (note that a role is required because we want to have a set of two rules):
+```
+{
+	"kind": "role",
+	"name": "deploymentConfigLabelers",
+	"namespace": "chisel",
+	"rules": [
+		{ "verbs": ["watch", "list", "get"], "resourceKinds": ["DeploymentConfig"] },
+		{ "verbs": ["update"], "resourceKinds": ["DeploymentConfig"], "attributeRestrictions" : {"fieldsMutatable": ["labels"]} }
+	]
+}
+{
+	"kind": "roleBinding",
+	"name": "DeploymentConfigLabelerBots",
+	"namespace": "chisel",
+	"roleName": "deploymentConfigLabelers",
+	"userNames": ["ProtectorBot", "DeprotectorBot"]
+}
+```
+
+Finally, `Hubert` wants to prevent `Curt` from ever deleting a protected deploymentConfig again, so he creates this role and roleBinding:
+```
+{
+	"kind": "role",
+	"name": "fatFingeredEditor",
+	"namespace": "chisel",
+	"rule": [
+		{ "Deny": "true", "Verbs": ["delete"], "ResourceKinds": ["DeploymentConfig"], "AttributeRestrictions" : {"labelsContain": ["protected"]} }
+	]
+}
+{
+	"kind": "roleBinding",
+	"name": "FatFingeredEditors",
+	"namespace": "chisel",
+	"roleName": "fatFingeredEditor",
+	"userNames": ["Curt"]
+}
+```
+
+## Basic Concepts
+ 1.  A PolicyRule expresses a permission containing: Deny, []Verb, []ResourceKinds (pod, deploymentConfig, *-(all kinds), etc)
+ 1.  A Role is a way to name a set of PolicyRules
+ 1.  A RoleBinding is a way to associate a single PolicyRule with a given user or team.  A RoleBinding references (but does not include) a Role in either the local namespace or the global namespace.
+ 1.  A global namespace exists.  This namespace is special.  Roles defined in the global namespace can be referenced from any other namespace.  Bindings defined in the global namespace apply to **all** namespaces (an admin in the global namespace can edit any resource in any namespace).  RoleBinding policy rules cannot be overridden by any local binding (a local PolicyBindingRule that denies actions to the global admin will be overriden by the binding defined in the global namespace).
+ 1.  A Policy is a container object that attempts to mitigate referential integrity problems.  Referential integrity problems **still exist!**  Local RoleBindings can still reference global Roles stored in a different object, so dangling references can still happen.  In usage, a RoleBinding that references a missing Role will produce an error.
+ 1.  Roles versus Teams.  A Role is a grouping of PolicyRules.  A team is a grouping of users.  A RoleBinding associates a Role with a Team.  This distinction makes Teams easier to share across projects.
+
+Policy Evaluation
+In order to determine whether a request is authorized, the AuthorizationAttributes are tested in the following order:
```

Comment:
> Would it be more efficient to evaluate in this order?

Yes, but it would eliminate feedback about being allowed and later denied.  I haven't dealt with questions about "why doesn't my policy work" before, but it seems like that would be useful information.  Not worth it?